File: codes/DSA 2023/reversing_string.cpp
#include <iostream>
#include <stack>
using namespace std;

void reverse_string(string s){
    stack<string> st;
    
    for(int i=0; i<s.length(); i++){
        string word ="";
         while(i<s.length() && s[i]!= ' '){
            word = word + s[i];
            i++;
         }
         st.push(word);
    }
    for(int i =0; i<s.length(); i++){
        cout<<st.top()<<" ";
        st.pop();
    }

}
int main(){
    string s =" how you doin' ";
    reverse_string(s);
     return 0;
}

File: codes/DSA 2023/reversing_string.py
def reverse_string(strng):
    
    stck =[]
    s = strng.split(" ")
    for i in s:
        stck.append(i)
        s=''
    while len(stck) != 0:
        
        s = s +" "+ stck[len(stck)-1]
        
        stck.pop()
    return s

print(reverse_string('how you doin !'))
print(type(reverse_string('how you doin !')))